simulator/event_manager.go

```diff
@@ -320,17 +320,46 @@ func (c *EventHistoryCollector) typeMatches(event types.BaseEvent, spec *types.E
 	return false
 }
 
+func (c *EventHistoryCollector) timeMatches(event types.BaseEvent, spec *types.EventFilterSpec) bool {
```

Comment:
Nit: In the future, if the number of matcherFn we want to support increases, instead of methods we could make these standalone functions (easier to unit test) using closures, e.g.

```go
type matchFn func(event types.BaseEvent, spec *types.EventFilterSpec) bool

func timeMatchFn(event types.BaseEvent, begin,end time.Time) matchFn {
    return func(event types.BaseEvent, spec *types.EventFilterSpec) bool {
        // logic
    }
}
```

And inject these matchFn when we create the collector(s).